codegen/src/main.zig

```diff
@@ -113,10 +118,15 @@ fn processDirectories(models_dir: std.fs.Dir, output_dir: std.fs.Dir) !void {
     defer manifest_file.close();
     const manifest = manifest_file.writer();
     var mi = models_dir.iterate();
+
+    const generating_models_progress = parent_progress.start("generating models", 400);
```

Comment:
Thanks for the PR. I'm merging the PR, but also I added a calculation for this in 6ed3a7e